crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -104,6 +113,18 @@ pub trait L2BlockSource: Send + Sync {
     async fn get_l2_block(&mut self, height: BlockHeight) -> Result<BlockInfo>;
 }
 
+#[derive(Debug, Default)]
+pub struct DaCommitDetails {
+    pub l2_block_range: core::ops::Range<u32>,
+    pub blob_bytes: u32,
+    pub blob_cost_wei: u32,
+    pub partial_block_heights: Option<[u32; 2]>,
+}
+
+pub trait DaCommitSource: Send + Sync {
+    fn get_da_commit_details(&mut self) -> Result<Option<DaCommitDetails>>;
```

Comment:
What's the reasoning behind the mutable receiver? The interface looks like we're just doing a lookup in which case we're probably not mutating anything. Or do we anticipate that the mutable receiver is needed?

Reply:
anticipated! this will most likely be changed in the follow up PR when we have something more concrete, I think its fine for now

Reply:
Fair enough, and yeah it is more conservative to require &mut self so it shouldn't be any problems relaxing it later if it's unused.